core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -429,6 +436,19 @@ public ContainerisedDockerCompose(List<File> composeFiles, String identifier) {
         addEnv("DOCKER_HOST", "unix:///docker.sock");
         setStartupCheckStrategy(new IndefiniteWaitOneShotStartupCheckStrategy());
         setWorkingDirectory(containerPwd);
+
+        String dockerConfigEnv = System.getenv(DOCKER_CONFIG_ENV);
+        String dockerConfigProperty = System.getProperty(DOCKER_CONFIG_PROPERTY);
+        Path dockerConfig = Paths.get(System.getProperty("user.home"), ".docker", "config.json");
+
+        if (dockerConfigEnv != null && !dockerConfigEnv.trim().isEmpty() && Files.exists(Paths.get(dockerConfigEnv))) {
+            addFileSystemBind(dockerConfigEnv, DOCKER_CONFIG_FILE, READ_ONLY);
+        } else if (dockerConfigProperty != null && !dockerConfigProperty.trim().isEmpty() && Files.exists(Paths.get(dockerConfigProperty))) {
+            addFileSystemBind(dockerConfigProperty, DOCKER_CONFIG_FILE, READ_ONLY);
+        } else if (Files.exists(dockerConfig)) {
+            addFileSystemBind(dockerConfig.toString(), DOCKER_CONFIG_FILE, READ_ONLY);
```

Comment:
I think the whole block should go into a private method `mountDockerConfig()`.

In addition I think this method could be structured a bit more concise, like maybe
```
String dockerConfigPath = determineDockerConfigPath();
addFileSystemBind(dockerConfigPath , DOCKER_CONFIG_FILE, READ_ONLY);
```